narwhal/worker/src/client.rs

```diff
@@ -0,0 +1,67 @@
+use std::sync::Arc;
+
+use arc_swap::ArcSwapOption;
+use thiserror::Error;
+use types::{metered_channel::Sender, Transaction, TxResponse};
+
+static LOCAL_NARWHAL_CLIENT: ArcSwapOption<LocalNarwhalClient> = ArcSwapOption::const_empty();
+
+/// The maximum allowed size of transactions into Narwhal.
+pub const MAX_ALLOWED_TRANSACTION_SIZE: usize = 6 * 1024 * 1024;
+
+/// Errors returned to clients submitting transactions to Narwhal.
+#[derive(Clone, Debug, Error)]
+pub enum NarwhalError {
+    #[error("Failed to include transaction in a header!")]
+    TransactionNotIncludedInHeader,
+
+    #[error("Narwhal is shutting down!")]
+    ShuttingDown,
+
+    #[error("Transaction is too large: size={0} limit={1}")]
+    TransactionTooLarge(usize, usize),
+}
+
+/// A client that connects to Narwhal locally.
+#[derive(Clone)]
+pub struct LocalNarwhalClient {
+    tx_batch_maker: Sender<(Transaction, TxResponse)>,
+}
+
+impl LocalNarwhalClient {
+    pub fn new(tx_batch_maker: Sender<(Transaction, TxResponse)>) -> Arc<Self> {
+        Arc::new(Self { tx_batch_maker })
+    }
+
+    /// Sets the global instance of LocalNarwhalClient.
+    pub fn set(instance: Arc<Self>) {
```

Comment:
just a thought-I might even name these `{set,get}_global` just to make it very clear in names what's happening?